### PR TITLE
Static head

### PR DESCRIPTION
**Pull request by IngelaLind**
Added components:
Boundaries.AccelerationBoundary; sets a global acceleration vector
Processes.StaticHead; adds or subtracts static head depending on acceleration and positions
Boundaries.TankCuboid; a tank for incompressible + gas media, which takes into account the global acceleration and the positions of an arbitrary number of inlets and outlets. This one has cuboid geometry and disregards acceleration in y-direction.
Boundaries.Internal.PartialTank; all the media related equations for a tank, setup to make it easy to model an arbitrary geometry by extension of this model
Boundaries.Tests; eight different test models for tanks and staticHead
Media.myMedia.Incompressible.Examples.Dowcal100; from public data
Media.myMedia.Incompressible.Examples.TestDowcal100
Media.myMedia.Incompressible.Examples.TestLiqLoopDowcal100
Media.myMedia.Incompressible.Examples.Dowcal100E; from public data
Media.myMedia.Incompressible.Examples.TestDowcal100E
Media.myMedia.Incompressible.Examples.JP8; Jet propulsion fuel
Media.myMedia.Incompressible.Examples.TestJP8
Media.myMedia.Incompressible.Examples.TestLiqLoopJP8
Media.myMedia.GasAndIncompressible.PartialGasAndIncompressible; base class to be able to combine e.g. fuel and air
Media.myMedia.GasAndIncompressible.JP8DryAir
Media.myMedia.GasAndIncompressible.Examples; four test models for JP8DryAir

### CODE REVIEW

**Comment by dzimmer** [comment]
First of all: Many thanks! That is quite a contribution. You really seem to have solved the general case.

- The main design choices meaning which components to add to the library seem correct to me. This all makes perfect sense.
- The value of having a medium model that handles gas and liquid state really shows here. It would be great to have water / air as a second implementation. 
- Minor cosmetics needed for the global acceleration model. setFromExpression serves no real purpose and can be removed. Icon needs a little improvement
- without further understanding I would prefer to remove the chaoticLife functionality at least for now. (may still be added later)

Do not take this comment as cause for any action. It primarily serves documenting my state of analysis. Certain modifications are also easier to do on our side.
Code analysis continues....

**Comment by dzimmer** [comment]
Further small issues:
- The code contains various fudge constants which should be turned into actual parameters / constants
- Some more documentation  and explanation needs to be added. I think, I understand the code now... 
- The authorship must be stated in the revision element of the models

I will do these changes and push to this pull-request repo branch.. Ingela, please make sure that I or contributors have rights to push on this branch. Thx

**Comment by dzimmer** [comment]
Ingela, I will proceed as detailed in:
https://gist.github.com/wtbarnes/56b942641d314522094d312bbaf33a81

but not today anymore...

**Comment by IngelaLind** [comment]
Dear Dirk,
I’m happy you liked my contribution.  I have no reservations on the changes you suggest. Do I understand you right that a fudge constant is something declared as Real x=2; without a constant or parameter declaration in front?
I am sure that more explanations are needed! It is just so hard to tell which ones, when those I work with have no trouble understanding the new parts I added, which they are familiar with from previous models, but still struggles with the basic concept of the library.
I totally forgot about authorship. Thanks for pointing that out.

It should be very easy to add air/water as a second GasAndIncompressible media model. Do you want me to do that?

I have been tinkering with the icon for the acceleration component the last weeks (the icon layer is REALLY hard to work on!) to add a dynamic indicator of the direction. This is a reasonable request from my colleagues. The directions of the acceleration is actually one of the things that is easiest to get wrong in day-to-day work. Do you want that now as well?

I have also added our logotype to this mail, if you want to add it to your contributors page.
Best regards,
Ingela

Ingela Lind, M Sc, Ph D
Technical Fellow
Simulation and Thermal Analysis
Vehicle Systems
SAAB Aerosystems

***@***.***
From: Dirk Zimmer ***@***.***>
Sent: den 7 juni 2024 19:06
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Further small issues:

  *   The code contains various fudge constants which should be turned into actual parameters / constants
  *   Some more documentation and explanation needs to be added. I think, I understand the code now...
  *   The authorship must be stated in the revision element of the models

I will do these changes and push to this pull-request repo branch.. Ingela, please make sure that I or contributors have rights to push on this branch. Thx

**Comment by dzimmer** [comment]
work to be continued. finished for today

**Comment by IngelaLind** [comment]
It is not as easy as I thought to add a water/air media since the incompressible media has no common base class with any of the water models. I had expected at least partialMedium in common. I do not know at the moment if it is easier to add another simplified water in the incompressible base class or do the better thing and make a GasAndCompressible partial class or rework the GasAndIncompressible to be more flexible. I do not have any motivation other than my own curiosity to do any of it and since we have no interest in water tanks at the moment,  it is hard to get that work funded.

**Comment by dzimmer** [comment]
Thanks for the warning. Our application would be to model a water extractor (and a corresponding water injector) A colleague of mine could attempt it but he is less familiar with Media modelling

**Comment by IngelaLind** [comment]
To do a good water extractor you would need to work with a media with the components dry air +water vapour + free water. That is more complex than the dry air + fuel mixture I did, where the fuel vapour is totally neglected. It is a very good idea to do one, though. We would certainly need it for our applications.

From: Dirk Zimmer ***@***.***>
Sent: den 13 juni 2024 20:06
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Thanks for the warning. Our application would be to model a water extractor (and a corresponding water injector) A colleague of mine could attempt it but he is less familiar with Media modelling

**Comment by dzimmer** [comment]
Alright. I have added a color scheme. We have used light green for all those components that use dynamic pressure. 
Now I have introduced brown/orange to mark components using the gravity field.
It remains an open design question whether to include the gravity field into the drops of common but (also to avoid conflicts with parallel developments), we will not do that now but may do so in a next version. Hence the warning in the doc of the components

**Comment by dzimmer** [comment]
Ingela, Can you check the simulation results on your side? My modifications should not have changed anything.
If you give thumbs up. I'll accept the Pull request

**Comment by dzimmer** [comment]
> To do a good water extractor you would need to work with a media with the components dry air +water vapour + free water. That is more complex than the dry air + fuel mixture I did, where the fuel vapour is totally neglected. It is a very good idea to do one, though. We would certainly need it for our applications.

A colleague of mine has implemented a water extractor volume using water and moist air as seperate media. However this has its complications when the tanks drains. I propose that you do both a common effort.

**Comment by IngelaLind** [comment]
Good. Then I should inform you that the present version of the acceleration boundary is simplified and not complete. For a large aircraft or other vessels moving in 3D, the rotation and center of rotation would also be needed. I do have the necessary equations somewhere, but it was not a priority now. So, the current version is about a third of a complete set of variables.

From: Dirk Zimmer ***@***.***>
Sent: den 14 juni 2024 17:53
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Alright. I have added a color scheme. We have used light green for all those components that use dynamic pressure.
Now I have introduced brown/orange to mark components using the gravity field.
It remains an open design question whether to include the gravity field into the drops of common but (also to avoid conflicts with parallel developments), we will not do that now but may do so in a next version. Hence the warning in the doc of the components

**Comment by IngelaLind** [comment]
Of course! But after an entire morning spent on git/gitHub trouble and reconfiguration,  I have run out of time. I will do the testing earliest next Monday.
---------

Ingela, Can you check the simulation results on your side? My modifications should not have changed anything.

File: ThermofluidStream/Boundaries/Tests/DynamicBoundaries.mo
Comment (tobolar) [inline ThermofluidStream/Boundaries/Tests/DynamicBoundaries.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Boundaries/Tests/PhaseSeperator.mo
Comment (tobolar) [inline ThermofluidStream/Boundaries/Tests/PhaseSeperator.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Boundaries/Tests/Reservoir.mo
Comment (tobolar) [inline ThermofluidStream/Boundaries/Tests/Reservoir.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Boundaries/Tests/Volumes.mo
Comment (tobolar) [inline ThermofluidStream/Boundaries/Tests/Volumes.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Processes/FlowResistance.mo
Comment (tobolar) [inline ThermofluidStream/Processes/FlowResistance.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Processes/Tests/ConvectivePipe.mo
Comment (tobolar) [inline ThermofluidStream/Processes/Tests/ConvectivePipe.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

File: ThermofluidStream/Processes/Tests/ConvectivePipe_serial.mo
Comment (tobolar) [inline ThermofluidStream/Processes/Tests/ConvectivePipe_serial.mo:1]: Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

**Pull request merged**